src/mbgl/annotation/annotation_manager.hpp

```diff
@@ -48,6 +54,9 @@ class AnnotationManager : private util::noncopyable {
     ShapeAnnotationImpl::Map shapeAnnotations;
     std::vector<std::string> obsoleteShapeAnnotationLayers;
     std::set<AnnotationTileMonitor*> monitors;
+
+    SpriteStore spriteStore;
```

Comment:
Wouldn't be easier to have these objects living the `MapContext` object? They would survive a style change anyway and would have a more clear scope. The crash I'm seeing on Linux is related to destruction order, not threading issues.

We are calling `glObjectStore.performCleanup()` on `MapContext::cleanup()` just before calling `~MapContext()` that will ultimately calls the `~MapData()` -> `~AnnotationManager()` -> `~SpriteAtlas()` that tries to dispose the textures but they never get cleaned up.

This patch fixes the crash on your branch: https://github.com/mapbox/mapbox-gl-native/commit/c15e1e6945dee649f2ef705211a9c22c1416854a